test/integration/roles/test_s3_bucket/tasks/main.yml

```diff
@@ -0,0 +1,254 @@
+---
+# tasks file for test_s3
+# OUTLINE:
+#   Test creating a bucket                                      (bucket1 created)
+#   Test creating a bucket with tags                            (bucket2 created)
+#   Test modifying tags of a bucket                             (bucket2 modified)
+#   Test creating a bucket with policy                          (bucket3 created)
+#   Test doing something not permitted by the policy            (bucket3 not modified)
+#   Test creating a bucket that already exists                  (bucket3 not modified)
+#   Test removing a policy from a bucket                        (bucket3 modified)
+#   Test deleting a bucket that doesn't exist                   (no bucket affected)
+#   Test deleting a bucket with an object in it without force   (bucket1 not modified)
+#   Test deleting a bucket with an object in it with force      (bucket1 deleted)
+#   Test deleting an empty bucket without force                 (bucket2 deleted)
+#   Test deleting an empty bucket with force                    (bucket3 deleted)
+# ============================================================
+- name: generate random name for the bucket name
+  shell: echo ansible_test_$RANDOM
+  register: bucket
+# ============================================================
+- name: create temporary file object to use later
+  shell: mktemp /tmp/XXXXXXXX
```

Comment:
I've gotten rid of shell module use and replaced it with tempfile because it seemed simpler. I don't know if that's OK - if you want me to use command I'll re-fix to use that.